requirements.txt

```diff
@@ -21,3 +21,4 @@ pyarrow >= 0.16
 responses >= 0.12.0
 urllib3 >= 1.25.10
 fastapi
+safir >= 3.4.0
```

Comment:
You also need to add this to pyproject.toml so that `pip install lsst-daf-butler` works. Ideally there would be an optional `server` definition in `pyproject.toml` so you can do `pip install lsst-daf-butler[server]`. If it's optional the test code will have to skip if safir is missing.